helper/controller/controller_test.go

```diff
@@ -26,7 +26,7 @@ func TestController_initialData(t *testing.T) {
 	require := require.New(t)
```

Comment:
These tests worked by exposing a `data` map that contained all the updates via the callbacks. To test the new `Delete()` functionality, I have exposed a new map called `deleted` that lists the keys that were deleted and their values so we can test that `Delete(key, obj)` was called as expected.